src/formatDate.js

```diff
@@ -8,7 +8,29 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const object = {};
+  const splitDate = date.split(fromFormat[fromFormat.length - 1]);
+  const result = [];
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    object[fromFormat[i]] = splitDate[i];
+  }
+
+  if (object.hasOwnProperty('YYYY')) {
+    object.YY = object.YYYY.slice(2);
+  }
+
+  if (object.YY >= 30) {
+    object.YYYY = '19' + object.YY;
+  } else {
+    object.YYYY = '20' + object.YY;
+  }
```

Comment:
The logic to convert from 'YY' to 'YYYY' should only be applied if 'YY' is present in the 'fromFormat' and 'YYYY' is required in the 'toFormat'. Additionally, the current implementation will overwrite the 'YYYY' value even if it was originally provided. Consider checking if 'YY' is in the 'fromFormat', if 'YYYY' is in the 'toFormat', and only then perform the conversion.